### ISSUE TITLE
CA1031 False Positive

### ISSUE THREAD

**Issue by RobSiklos** [dotnet/roslyn-analyzers#2365]
#### Analyzer package

[Microsoft.CodeAnalysis.FxCopAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.FxCopAnalyzers)

#### Package Version

v2.9.2

#### Diagnostic ID

[CA1031](https://docs.microsoft.com/en-us/visualstudio/code-quality/ca1031-do-not-catch-general-exception-types)

#### Repro steps

```
public static class Class1
{
    public static void Foo()
    {
        try
        {
        }
        catch (Exception ex)
        {
            throw new InvalidOperationException("The Foo operation failed. If the problem persists, please contact your administrator.", ex);
        }
    }
}
```
#### Expected behavior
No violation, since the Exception is included as the inner exception of the thrown exception.

#### Actual behavior
CA1031: Modify 'Foo' to catch a more specific exception type, or rethrow the exception.

**Comment by RobSiklos** [comment]
Sorry - looks like this is a dupe of issue #2278
